packages/visual-editor/src/components/Dropzone/useComponent.ts

```diff
@@ -67,30 +69,37 @@ export const useComponent = ({
     areEntitiesFetched,
     resolveDesignValue,
     renderDropzone,
-    definition: componentRegistration.definition,
+    definition: componentRegistration?.definition,
     userIsDragging,
   });
 
+  const createElementToRender = (componentRegistration: ComponentRegistration) => {
+    return builtInComponents.includes(node.data.blockId || '')
+      ? (dragProps?: NoWrapDraggableProps) =>
+          React.createElement(componentRegistration.component, { ...dragProps, ...componentProps })
+      : node.type === ASSEMBLY_NODE_TYPE
+        ? // Assembly.tsx requires renderDropzone and editorMode as well
+          () => React.createElement(componentRegistration.component, componentProps)
+        : () =>
+            React.createElement(
+              ImportedComponentErrorBoundary,
+              null,
+              React.createElement(componentRegistration.component, otherComponentProps),
+            );
```

Comment:
As an alternative to the method you already have, I think the desired functionality could also be done here by adding another elseif check if the definition is missing, and if so, render the SelectableComponent instead of the componentRegistration.component.

I think if you did so, the SelectableComponent wouldn't need any of the complexity it currently has to support being a draggable, as it would get wrapped in the DraggableComponent just like any other component.

Reply:
Can I please ask to break this one liner into a more human readable form? As much as one liners are fun, I'd prefer having ifs and proper intermediate steps named well with variables to help understand it better.

Also by writing this code this way you are promoting the rest of the devs in future (who might need to edit it) to follow this style.